src/app/features/pdf-generator/pdf-generator-api/pdf-generator-api.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Inject, Injectable } from "@angular/core";
+import { FileService } from "../../file/file.service";
+import { SafeUrl } from "@angular/platform-browser";
+import { Entity } from "app/core/entity/model/entity";
+import { Observable, throwError } from "rxjs";
+import { EntityMapperService } from "../../../core/entity/entity-mapper/entity-mapper.service";
+import { EntityRegistry } from "../../../core/entity/database-entity.decorator";
+import { SyncStateSubject } from "../../../core/session/session-type";
+import { HttpClient } from "@angular/common/http";
+import { NotAvailableOfflineError } from "../../../core/session/not-available-offline.error";
+import { NAVIGATOR_TOKEN } from "../../../utils/di-tokens";
+import { switchMap } from "rxjs/operators";
+import { FileTemplate } from "../file-template.entity";
+
+interface TemplateUploadResponseDto {
+  templateId: string;
+}
+
+interface TemplateRenderRequestDto {
+  convertTo: string;
+  reportName: string;
+  data: Object;
+}
+
+/**
+ * Interact with the PDF Template Generation API that uses File Templates to generate custom pdf documents.
+ */
+@Injectable({
+  providedIn: "root",
+})
+export class PdfGeneratorApiService extends FileService {
+  readonly BACKEND_URL = "/query/api/v1/export/";
+
+  constructor(
+    entityMapper: EntityMapperService,
+    entities: EntityRegistry,
+    syncState: SyncStateSubject,
+    @Inject(NAVIGATOR_TOKEN) private navigator: Navigator,
+    private httpClient: HttpClient,
+  ) {
+    super(entityMapper, entities, syncState);
+  }
+
+  /*
+  --- FileService methods ---
+  */
+
+  /**
+   * Upload a new template file to the API.
+   * @param file to be uploaded to the API as a template
+   * @param entity
+   * @param property
+   * @return The template ID generated by the API
+   */
+  uploadFile(file: File, entity: Entity, property: string): Observable<string> {
+    if (!this.navigator.onLine) {
+      return throwError(() => new NotAvailableOfflineError("File Attachments"));
+    }
+
+    const formData = new FormData();
+    formData.append("template", file, file.name);
+
+    return this.httpClient.post(this.BACKEND_URL + "template", formData).pipe(
+      switchMap(async (res: TemplateUploadResponseDto) => {
+        // TODO: maybe store template filename + templateId (API) in entity, to show the user better meta-information?
```

Comment:
The filename of the rendered file can also be defined via variables (reportName). There is a "Content-Disposition" Header from carbone.io, wich we can forward to the frontend.